src/python/qeqiskit/backend/backend.py

```diff
@@ -78,101 +80,196 @@ def run_circuit_and_measure(self, circuit, **kwargs):
         Returns:
             a list of bitstrings (a list of tuples)
         """
-        super().run_circuit_and_measure(circuit)
-        num_qubits = len(circuit.qubits)
 
-        ibmq_circuit = circuit.to_qiskit()
-        ibmq_circuit.barrier(range(num_qubits))
-        ibmq_circuit.measure(range(num_qubits), range(num_qubits))
-
-        # Run job on device and get counts
-        raw_counts = (
-            execute(
-                ibmq_circuit,
-                self.device,
-                shots=self.n_samples,
-                optimization_level=self.optimization_level,
-            )
-            .result()
-            .get_counts()
-        )
+        return self.run_circuitset_and_measure([circuit], **kwargs)[0]
 
-        if self.readout_correction:
-            raw_counts = self.apply_readout_correction(raw_counts, kwargs)
-
-        # qiskit counts object maps bitstrings in reversed order to ints, so we must flip the bitstrings
-        reversed_counts = {}
-        for bitstring in raw_counts.keys():
-            reversed_counts[bitstring[::-1]] = int(raw_counts[bitstring])
-        measurements = Measurements.from_counts(reversed_counts)
-
-        return measurements
-
-    def run_circuitset_and_measure(self, circuitset, **kwargs):
-        """Run a set of circuits and measure a certain number of bitstrings.
-        Note: the number of bitstrings measured is derived from self.n_samples
+    def expand_circuitset(
```

Comment:
This function name is slightly confusing for me.
What about sth more like `transform_circuitset_to_fit_backend_requirements` (not that this name is particularly good).

Reply:
I changed it to `transform_circuitset_to_ibmq_experiments`, how about that? (Qiskit refers to circuits to be submitted to a backend as `experiments`.)